core/transformer/src/IModelTransformer.ts

```diff
@@ -1146,76 +1131,123 @@ export class IModelTransformer extends IModelExportHandler {
+   * again but the remapping state will cause already mapped elements to be skipped.
+   * To "resume" an iModel Transformation you need:
+   * - the sourceDb at the same changeset
+   * - the same targetDb in the state in which it was before
+   * @param statePath the path to the serialized state of the transformer, use [[IModelTransformer.saveStateToFile]] to get this from an existing transformer instance
+   * @param constructorArgs remaining arguments that you would normally pass to the Transformer subclass you are using, usually (sourceDb, targetDb)
+   * @note custom transformers with custom state may need to override this method in order to handle loading their own custom state somewhere
+   */
+  public static resumeTransformation<SubClass extends new(...a: any[]) => IModelTransformer = typeof IModelTransformer>(
+    this: SubClass,
+    statePath: string,
+    ...constructorArgs: ConstructorParameters<SubClass>
+  ): InstanceType<SubClass> {
+    const transformer = new this(...constructorArgs);
+    const db = new SQLiteDb();
+    db.openDb(statePath, OpenMode.Readonly);
+    transformer.loadStateFromDb(db);
     db.closeDb();
     return transformer as InstanceType<SubClass>;
   }
 
   /**
-   * Serialize the state of the active transformation to a file path, if a file at the path already exists, it will be overwritten
-   * This state can be used by [[resumeTransformation]] to resume a transformation from this point.
-   * The serialization format is a custom sqlite database
-   * @note custom transformers with custom state may need to override this method in order to handle storing their own custom state somewhere,
-   *       potentially inside the same sqlite file in separate tables
+   * You may override this to store arbitrary json state in a transformer state dump, useful for some resumptions
+   * @see [[IModelTransformer.saveStateToFile]]
    */
-  public saveStateToFile(nativeStatePath: string): void {
-    const db = new SQLiteDb();
-    if (IModelJsFs.existsSync(nativeStatePath))
-      IModelJsFs.unlinkSync(nativeStatePath);
-    db.createDb(nativeStatePath);
+  protected getAdditionalStateJson(): any {
+    return {};
+  }
+
+  /**
+   * You may override this to load arbitrary json state in a transformer state dump, useful for some resumptions
+   * @see [[IModelTransformer.loadStateFromFile]]
+   */
+  protected loadAdditionalStateJson(_additionalState: any): void {}
+
+  /**
+   * Save the state of the active transformation to an open SQLiteDb
+   * You can override this if you'd like to write custom tables to the resumable dump state, but you should call
+   * this super implementation
+   * @note the SQLiteDb must be open
+   */
+  protected saveStateToDb(db: SQLiteDb): void {
     const jsonState: TransformationJsonState = {
+      transformerClass: this.constructor.name,
```

Comment:
I can think of some ways in which `constructor.name` won't suffice (import IModelTransformer using an alias and define a subclass called IModelTransformer; instantiate IModelTransformer then add properties and replace methods) but people would have to go out of their way to break it.